src/picterra/client.py

```diff
@@ -39,7 +41,10 @@ def validate_detector_args(detection_type: str, output_type: str, training_steps
 
 class APIClient():
     """Main client class for the Picterra API"""
-    def __init__(self, api_key=None, base_url=None):
+    def __init__(
+        self, api_key: str = '', base_url: str = '',
+        timeout: int = 5, max_retries: int = 5, backoff_factor: int = 10
```

Comment:
5s seems pretty low, I'd put something like 30 for the default. Also maybe max_retries down to 3 to avoid getting hammered by retries ?